src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -198,95 +151,147 @@ vtkSmartPointer< vtkUnstructuredGrid > loadVTKMesh( Path const & filePath )
+ * @param numParts number of partitions (cuts)
+ * @param numRanks number of MPI ranks
+ * @return a vector of rank assignments
+ *
+ * This function merges partitions starting from the back, until the desired number is reached.
+ * It is a stripped down version of vtkDIYKdTreeUtilities::ComputeAssignments(). We cannot call
+ * that function directly because it's part of a VTK private module (i.e. header not installed).
+ * We rely on copying the assignment procedure for lack of a better solution.
+ * In particular, there is no API to extract this information out of vtkRedistributeDataSetFilter.
+ */
+std::vector< int > computePartitionAssignments( int const numParts, int const numRanks )
+{
+  GEOSX_ERROR_IF_NE_MSG( numParts, vtkMath::NearestPowerOfTwo( numRanks ),
+                         "Number of partitions is not a nearest power of 2 of number of ranks."
+                         "This could indicate a change in VTK implementation of mesh partitioner." );
+
+  std::vector< int > assignments( numParts );
+  std::iota( assignments.begin(), assignments.end(), 0 );
+
+  int numToMerge = numParts - numRanks;
+  int rank = numRanks - 1;
+  for( int cc = numParts - 2; cc >= 0 && numToMerge > 0; cc -= 2, --numToMerge, --rank )
+  {
+    assignments[cc] = assignments[cc + 1] = rank;
+  }
+  return assignments;
+}
+
+
 /**
  * @brief Redistribute the mesh among the available MPI ranks
  * @details this method will also generate global ids for points and cells in the VTK Mesh
  * @param[in] loadedMesh the mesh that was loaded on one or several MPI ranks
  * @param[out] cuts the bounding boxes used by the VTK partitioner needed to compute neighboring ranks
  */
-vtkSmartPointer< vtkUnstructuredGrid > redistributeMesh( vtkUnstructuredGrid & loadedMesh,
-                                                         std::vector< vtkBoundingBox > & cuts )
+vtkSmartPointer< vtkUnstructuredGrid >
+redistributeMesh( vtkUnstructuredGrid & loadedMesh,
+                  std::vector< vtkBoundingBox > & cuts )
 {
+  // Generate global IDs for vertices and cells
+  vtkNew< vtkGenerateGlobalIds > generator;
+  generator->SetInputDataObject( &loadedMesh );
+  generator->Update();
+
   // Redistribute data all over the available ranks
   vtkNew< vtkRedistributeDataSetFilter > rdsf;
-  rdsf->SetInputDataObject( &loadedMesh );
+  rdsf->SetInputDataObject( generator->GetOutputDataObject( 0 ) );
   rdsf->SetNumberOfPartitions( MpiWrapper::commSize() );
-  rdsf->GenerateGlobalCellIdsOn();
   rdsf->Update();
 
   cuts = rdsf->GetCuts();
-
-  // Generate global IDs for vertices and cells
-  vtkNew< vtkGenerateGlobalIds > generator;
-  generator->SetInputDataObject( vtkUnstructuredGrid::SafeDownCast( rdsf->GetOutputDataObject( 0 ) ) );
-  generator->Update();
-  vtkSmartPointer< vtkUnstructuredGrid > mesh = vtkUnstructuredGrid::SafeDownCast( generator->GetOutputDataObject( 0 ) );
-  return mesh;
+  return vtkUnstructuredGrid::SafeDownCast( rdsf->GetOutputDataObject( 0 ) );
```

Comment:
The order of filters is now swapped, because `vtkRedistributeDataSetFilter` in the latest VTK release (9.1.0) relies on presence of global point IDs to merge duplicate nodes when "gluing" together mesh partitions assigned to the same rank (it can also do it by comparing coordinates, but this is easier).